packages/babel-parser/src/parser/expression.js

```diff
@@ -1672,12 +1741,27 @@ export default class ExpressionParser extends LValParser {
     const containsEsc = this.state.containsEsc;
     this.parsePropertyName(prop, /* isPrivateNameAllowed */ false);
 
-    if (!isPattern && !containsEsc && !isGenerator && this.isAsyncProp(prop)) {
-      isAsync = true;
-      isGenerator = this.eat(tt.star);
-      this.parsePropertyName(prop, /* isPrivateNameAllowed */ false);
-    } else {
-      isAsync = false;
+    if (
+      !isPattern &&
+      !isGenerator &&
+      !containsEsc &&
+      this.maybeAsyncOrAccessorProp(prop)
+    ) {
+      // https://tc39.es/ecma262/#prod-AsyncMethod
+      // https://tc39.es/ecma262/#prod-AsyncGeneratorMethod
+      if (prop.key.name === "async" && !this.hasPrecedingLineBreak()) {
```

Comment:
Parsing `async` and `set`, `getter` have been unified in `parsePropertyDefinition` -- so they can share same sanity checks. Thus we don't need passing `containsEsc` (A tokenizing level states) to `parseObjectMethod`, instead `isAccessor` is pass.